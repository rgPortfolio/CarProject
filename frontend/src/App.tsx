import { RouteComponentProps, Router } from '@reach/router';
import './App.css';
import { Dashboard } from './Components/Dashboard';

let Home = () => <div>Home</div>
let Dash = () => <div>Dash</div>

function App() {
  return (
    <Router>
    <RouterPage path="/" pageComponent={<Dashboard />}/>
    {/* <RouterPage path="cars/:carId" pageComponent={<CarSelect />}/> */}
    </Router>
  );
}

const RouterPage = (
  props: { pageComponent: JSX.Element } & RouteComponentProps
) => props.pageComponent;

export default App;
