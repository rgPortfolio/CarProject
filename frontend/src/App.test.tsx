import App from './App';

import Enzyme, { shallow } from 'enzyme';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
