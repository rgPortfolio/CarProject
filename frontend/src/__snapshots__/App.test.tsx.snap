// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Layout>
  <Router>
    <RouterPage
      pageComponent={<CarSelect />}
      path="/"
    />
    <RouterPage
      pageComponent={<CarProfile />}
      path="cars/:id"
    />
  </Router>
</Layout>
`;
